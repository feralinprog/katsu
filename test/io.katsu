use: {
    "core.builtin.misc"
    "core.dynamic-variable"
    "core.io"
    "core.io.linux.file"
    "core.io.stream"
}

with-io: [
    try: [
        %open: "file-does-not-exist" flags: O_RDONLY
    ] except: {
        IOError, \c [
            print: "got IO error: " ~ c .message
        ]
    }

    with-input-stream: (open-input: "test/files/file1.txt") do: [
        let: s = read-all byte-array>string
        print: s
    ]

    # Do it again, but with a ridiculously small group size for reading. Make sure each group
    # is incorporated correctly.
    with-input-stream: (open-input: "test/files/file1.txt") do: [
        let: s = (read-all/group: 1 like: *input-stream* get buffer-exemplar) byte-array>string
        print: s
    ]

    let: nl = " "
    nl unsafe-write-u8-at-offset: 16 value: 10

    print: "writing to stdout:"
    write-from: ("abcde" ~ nl) string>byte-array

    print: "writing to stderr:"
    with-output-stream*: *error-stream* get do: [
        write-from: ("fake-err" ~ nl) string>byte-array
    ]
]