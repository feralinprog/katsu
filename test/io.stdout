got IO error: IO error (ENOENT): No such file or directory
this is a test file
with some test content
for a test program

this is a test file
with some test content
for a test program

writing to stdout:
abcde
writing to stderr:
